pkg/resource/managed_reconciler_test.go

```diff
@@ -0,0 +1,286 @@
+/*
+Copyright 2019 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package resource
+
+import (
+	"context"
+	"testing"
+
+	"k8s.io/apimachinery/pkg/runtime"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/pkg/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	"github.com/crossplaneio/crossplane/pkg/apis/core/v1alpha1"
+	"github.com/crossplaneio/crossplane/pkg/test"
+)
+
+var _ reconcile.Reconciler = &ManagedReconciler{}
+
+// Scenarios: (TODO:delete this block after impl is done)
+// create external from managed
+// update external with change from managed
+// delete managed and then purge external with ReclaimDelete: + check finalizer scenario, deletion should be done after a few calls
+// delete managed and see if external is touched with ReclaimRetain
+
+func TestManagedReconciler_Reconcile(t *testing.T) {
```

Comment:
This is purely aesthetics, but I prefer to avoid underscores in test names so that everything reads like `SomeCamelCaseTestName/SomeCamelCaseSubTest` in the unit test output.

Given the `ManagedReconciler` and `ClaimReconciler` only have one method each, what do you think about naming their tests `TestManagedReconciler` and `TestClaimReconciler`?